hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -1031,6 +1031,40 @@ public static long getMultipartSizeProperty(Configuration conf,
     return partSize;
   }
 
+  /**
+   * Validates the output stream configuration
+   * @param conf : configuration object for the given context
+   * @throws IOException : throws an IOException on config mismatch
+   */
+  public static void validateOutputStreamConfiguration(Configuration conf) throws IOException {
+    if(!checkDiskBuffer(conf)){
+      throw new IOException("Unable to create OutputStream with the given"
+          + " multipart upload and buffer configuration.");
+    }
+  }
+
+  /**
+   * Check whether the configuration for S3ABlockOutputStream is
+   * consistent or not. Multipart uploads allow all kinds of fast buffers to
+   * be supported. When the option is disabled only disk buffers are allowed to
+   * be used as the file size might be bigger than the buffer size that can be
+   * allocated.
+   * @param conf : configuration object for the given context
+   * @return true if the disk buffer and the multipart settings are supported
+   */
+  public static boolean checkDiskBuffer(Configuration conf) {
+    boolean isMultipartUploadEnabled = conf.getBoolean(MULTIPART_UPLOADS_ENABLED,
+        MULTIPART_UPLOAD_ENABLED_DEFAULT);
+    if (isMultipartUploadEnabled) {
+      return true;
+    } else if (!isMultipartUploadEnabled && conf.get(FAST_UPLOAD_BUFFER)
```

Comment:
can be simplified to
```
return isMultipartUploadEnabled
 || FAST_UPLOAD_BUFFER_DISK.equals(conf.get(FAST_UPLOAD_BUFFER, DEFAULT_FAST_UPLOAD_BUFFER));
```
that default in conf.get is critical to prevent NPEs if the option is unset, moving the constant first even more rigorous